comp/networkdeviceconfig/impl/networkdeviceconfig.go

```diff
@@ -27,24 +30,89 @@ type Provides struct {
 }
 
 type networkDeviceConfigImpl struct {
-	config config.Component
-	log    log.Component
+	config        *ProcessedNcmConfig
+	log           log.Component
+	clientFactory RemoteClientFactory
 }
 
 // NewComponent creates a new networkdeviceconfig component
 func NewComponent(reqs Requires) (Provides, error) {
+	ncmConfig, err := newConfig(reqs.Config)
+	if err != nil {
+		return Provides{}, reqs.Logger.Errorf("Failed to read network device configuration: %v", err)
+	}
 	impl := &networkDeviceConfigImpl{
-		config: reqs.Config,
-		log:    reqs.Logger,
+		config:        ncmConfig,
+		log:           reqs.Logger,
+		clientFactory: &SSHClientFactory{},
 	}
 	provides := Provides{
 		Comp: impl,
 	}
 	return provides, nil
 }
 
-// RetrieveConfiguration retrieves the configuration for a given network device ID
-func (n networkDeviceConfigImpl) RetrieveConfiguration(_ string) (string, error) {
-	//TODO implement me
-	return "", nil
+// RetrieveRunningConfig retrieves the running configuration for a given network device IP
+func (n networkDeviceConfigImpl) RetrieveRunningConfig(ipAddress string) (string, error) {
+	commands := []string{
+		`show running-config`,
+	}
+	return n.retrieveConfiguration(ipAddress, commands)
+}
+
+// RetrieveStartupConfig retrieves the startup configuration for a given network device IP
+func (n networkDeviceConfigImpl) RetrieveStartupConfig(ipAddress string) (string, error) {
+	commands := []string{
+		`show startup-config`,
+	}
+	return n.retrieveConfiguration(ipAddress, commands)
+}
+
+// retrieveConfiguration retrieves the configuration for a given network device IP
+func (n networkDeviceConfigImpl) retrieveConfiguration(ipAddress string, commands []string) (string, error) {
+	deviceConfig, ok := n.config.Devices[ipAddress]
+	if !ok {
+		return "", n.log.Errorf("No authentication credentials found for device %s", ipAddress)
+	}
+	client, err := n.clientFactory.Connect(ipAddress, deviceConfig.Auth)
+	if err != nil {
+		return "", n.log.Errorf("Failed to connect to host %s: %v", ipAddress, err)
+	}
+	defer client.Close()
+
+	result := []string{}
+
+	for _, cmd := range commands {
+		session, err := client.NewSession()
+		if err != nil {
+			return "", n.log.Errorf("Failed to create session to device %s: %s", ipAddress, err)
+		}
+		n.log.Debugf("Running command: %s\n", cmd)
+		output, err := session.CombinedOutput(cmd)
+		if err != nil {
+			session.Close()
+			return "", n.log.Errorf("Command %s on device %s failed: %s\n", cmd, ipAddress, err)
+		}
+		n.log.Debugf("Output: %s\n", output)
```

Comment:
I think we should remove this for now and add it later when we have the config parsing/scrubbing since this output might have credentials/certifcats/etc